### PR TITLE
handle HTTP requests to "/" and "/api/v1.0/"

### PR DESCRIPTION
**Pull request by ktdreyer**
Return a small JSON response with links to the endpoints under `/` and `/api/v1.0/`.

The purpose of this change is to make the API more discoverable to clients    who are browsing around.

### CODE REVIEW

File: product_listings_manager/tests/test_root.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from product_listings_manager.app import app
+
+
+@pytest.yield_fixture
+def client():
+    client = app.test_client()
+    yield client
+
+
+class TestAbout(object):
```
Comment (simozhan) [inline product_listings_manager/tests/test_root.py]: I think this class name should be TestRoot, there is no about link in this page

File: product_listings_manager/tests/test_rest_api_v1.py [hunk 2]
```diff
@@ -4,22 +4,25 @@
...
-    Response.get_json = get_json
-
 
 @pytest.yield_fixture
 def client():
     client = app.test_client()
     yield client
 
 
+class TestRoot(object):
```
Comment (simozhan) [inline product_listings_manager/tests/test_rest_api_v1.py]: If call this TestRoot, it should be Api root, maybe TestApiRoot?

Comment (ktdreyer) [reply product_listings_manager/tests/test_rest_api_v1.py]: Good point. I've changed this to `TestIndex`, to match the new `Index` class in `rest_api_v1.py`

File: product_listings_manager/tests/test_root.py (hunk 1 above)
Comment (ktdreyer) [reply product_listings_manager/tests/test_root.py]: Yes! I made a copy & paste mistake here. I've amended this to be `TestRoot` now.

**Comment by ktdreyer** [comment]
Thanks for the review @simozhan . I've pushed a new version with your suggestions.

Would you please say more about the problem you're seeing with the links changing to "..." ?

File: product_listings_manager/tests/test_rest_api_v1.py [hunk 3]
```diff
@@ -4,22 +4,25 @@
...
     yield client
 
 
+class TestIndex(object):
+    def test_get_index(self, client):
+        r = client.get('/api/v1.0/')
+        expected_json = {
+            'about_url': 'http://localhost/api/v1.0/about',
+            'product_info_url': 'http://localhost/api/v1.0/product-info/:label',    # NOQA: E501
+            'product_listings_url': 'http://localhost/api/v1.0/product-listings/:label/:build_info'  # NOQA: E501
```
Comment (hluk) [inline product_listings_manager/tests/test_rest_api_v1.py]: You can remove these NOQAs, since #32 was merged.

Comment (ktdreyer) [reply product_listings_manager/tests/test_rest_api_v1.py]: Good call, done.

**Pull request merged**